### PR TITLE
Auto setup with carrier_add

### LINKED ISSUE #26: Feature: New carrier setup with m.carrier_add
m.carrier_add presently adds an FC to the internal database. However the information supplied could be used to automate the setup of Discord owner role ("P.T.N. NEWCARRIER (REG-NUM)"), crew role ("CREW: Newcarrier") and channel creation ("#ptn-newcarrier") which are all presently done manually. Channel would have to be created before ID is scraped for database but this would be helpful in preventing errors anyway.

If used in concert with issue #25 (crew roles command) would fully automate new carrier entry.

### PR DESCRIPTION
**Pull request by cimspin**
Implemented basic carrier channel and role setup to carrier_add. Closes #26. 

@gcruicks This should work but merging may create conflicts with other changes currently in PR. Note it required adding two new fields to the carrier db - crewrole and roleid. These will be used when implementing #25

### CODE REVIEW

File: MissionAlertBot.py [hunk 1]
```diff
@@ -1032,6 +1035,16 @@ def validate_response(react, user):
             break
 
 
+@bot.command(name='test')
+async def test(ctx, long_name):
+    strip1_name = long_name.replace(' ', '-')
+    await ctx.send(strip1_name)
+    strip2_name = strip1_name.replace('.', '')
+    await ctx.send(strip2_name)
```
Comment (gcruicks) [inline MissionAlertBot.py:1043]: You can chain string things.

`long_name.replace(' ', '-').replace('.', '')`

File: MissionAlertBot.py [hunk 2]
```diff
@@ -1043,23 +1056,51 @@ def validate_response(react, user):
...
+async def carrier_add(ctx, short_name, long_name, carrier_id):
     backup_database('carriers')  # backup the carriers database before going any further
 
-    print(f'Looking for discord channel to add the carrier: {discord_channel}')
+    # first create the new carrier's channel
+
+    # check whether channel already exists by sanitising the carrier's name input to match discord channel format, otherwise create one
+
+    strip1_name = long_name.replace(' ', '-')
+    strip2_name = strip1_name.replace('.', '')
```
Comment (gcruicks) [inline MissionAlertBot.py:1067]: This can be one line

File: MissionAlertBot.py [hunk 3]
```diff
@@ -1043,23 +1056,51 @@ def validate_response(react, user):
...
+    channel = discord.utils.get(ctx.guild.channels, name=strip2_name.lower())
+
+    if channel:
+        await ctx.send('Channel creation skipped: a channel already exists with this carrier\'s name')
```
Comment (gcruicks) [inline MissionAlertBot.py:1071]: Use " then you can use ' inside the string without escaping it. It doesn't change anything, just helps you avoid unnecessary escapes

File: MissionAlertBot.py [hunk 4]
```diff
@@ -1043,23 +1056,51 @@ def validate_response(react, user):
...
+        category = discord.utils.get(ctx.guild.categories, name="Drydock")
+        channel = await ctx.guild.create_text_channel(long_name, category=category)
+        print(f"Created {channel}")
+
     print(f'Channels: {ctx.guild.channels}')
-    channel = discord.utils.get(ctx.guild.channels, name=discord_channel)
+    
     if not channel:
-        raise EnvironmentError('Channel does not exist, go make it first and try again')
+        raise EnvironmentError('Could not create carrier channel')
```
Comment (gcruicks) [inline MissionAlertBot.py:1082]: I would add here the channel name to the return message

File: MissionAlertBot.py [hunk 5]
```diff
@@ -1032,6 +1035,16 @@ def validate_response(react, user):
             break
 
 
+@bot.command(name='test')
```
Comment (gcruicks) [inline MissionAlertBot.py:1038]: Should we remove this?

File: MissionAlertBot.py [hunk 6]
```diff
@@ -16,6 +16,7 @@
 import asyncpraw
 import asyncio
 import shutil
+import re
```
Comment (gcruicks) [inline MissionAlertBot.py:19]: Do you ever use re?

File: MissionAlertBot.py (hunk 5 above)
Comment (cimspin) [reply MissionAlertBot.py]: Yes!

File: MissionAlertBot.py (hunk 6 above)
Comment (cimspin) [reply MissionAlertBot.py]: No!

File: MissionAlertBot.py (hunk 1 above)
Comment (cimspin) [reply MissionAlertBot.py]: That is revolutionary

**Review by gcruicks — APPROVED** [review]

**Comment by gcruicks** [comment]
@cimspin - Best double check the extra change I made on top.

**Pull request merged**